core-common/src/main/java/org/mapstruct/MapMapping.java

```diff
@@ -55,10 +55,21 @@
     String valueDateFormat() default "";
 
     /**
-     * A key value qualifier can be specified to aid the selection process of a suitable mapper. This is useful in
-     * case multiple mappers (hand written of internal) qualify and result in an 'Ambiguous mapping methods found'
-     * error.
+     * A qualifier can be specified to aid the selection process of suitable {@link BeforeMapping} /
+     * {@link AfterMapping} methods.
+     * <p>
+     * Use {@link #keyQualifiedBy()} / {@link #valueQualifiedBy()} to specify qualifiers to be used for selecting key /
+     * value mapping methods.
      *
+     * @return the qualifiers
+     */
+    Class<? extends Annotation>[] qualifiedBy() default { };
```

Comment:
Yea, I know what you mean. I added this here for completeness, as you wouldn't have any other way to use qualifiers for lifecycle methods together with map-mapping methods. There is no other application for this `qualifiedBy`, yet. 
One could think of selecting specific factory methods for initializing the map type to return (e.g. `@TinyMap`, or `@HugeMap`, or `@SynchronizedMap`, ...). If you think it's too much for now, I'll just drop it out again.